sbi/utils/sbiutils.py

```diff
@@ -79,14 +79,58 @@ def clamp_and_warn(name: str, value: float, min_val: float, max_val: float) -> f
     return clamped_val
 
 
+def z_score_parser(z_score_flag: Optional["str"]) -> Tuple[bool, bool]:
+    """Parses string z-score flag into booleans.
+
+    Converts string flag into booleans denoting whether to z-score or not, and whether
+    data dimensions are structured or independent.
+
+    Args:
+        z_score_flag: str flag for z-scoring method stating whether the data
+        dimensions are "structured" or "independent", or does not require z-scoring
+        ("none" or None).
+
+    Returns:
+        Flag for whether or not to z-score, and whether data is structured
+    """
+    if type(z_score_flag) is bool:
+        # Raise warning if boolean was passed.
+        warnings.warn(
+            """Boolean flag for z-scoring is accepted for backwards compatibility only.
```

Comment:
Boolean flag for z-scoring is deprecated as of sbi v0.18.0. It will be removed in a future release. Use 'none', 'independent', or 'structured' to indicate z-scoring option.